### PR TITLE
Fixing "PHP Notice - Trying to access array offset on value of type null"

### PR DESCRIPTION
**Pull request by escopecz**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes one issue reported in https://github.com/mautic/mautic/pull/10883 and Acquia's https://backlog.acquia.com/browse/MAUT-4567

#### Description:

This fixes a PHP notice when opening a custom object campaign condition. It is not visible in the production but blocks developers or testers on dev environment.

#### Steps to test this PR:

1. Create a custom object with a field
2. Create a new campaign
3. Select any source
4. Try to create a condition based on the custom object created in step 1.

If you are using dev environment (index_dev.php) then the form won't even load. If you are on production environment then the PHP notice will be logged into the logs.

Once you apply this PR then there will be no PHP notice.

### CODE REVIEW

**Review by rohitp19 — DISMISSED** [review]
Thanks @escopecz

**Comment by RCheesley** [comment]
Hi folks! Are we able to get this merged? Or is it blocked on something?

**Comment by escopecz** [comment]
The bot does not want to choose the second reviewer, so I requested Fedy :)

File: Form/Type/CampaignConditionFieldValueType.php
Comment (fedys) [inline Form/Type/CampaignConditionFieldValueType.php]:
```suggestion
        if (isset($options['data']['field']) && isset($fields[$options['data']['field']])) {
```
This might be simplified a bit.

**Review by fedys — CHANGES_REQUESTED** [review]
See my code suggestion.

File: Form/Type/CampaignConditionFieldValueType.php
Comment (escopecz) [reply Form/Type/CampaignConditionFieldValueType.php]: That's what I thought too, but I was still getting the PHP notice. So I added `isset($options['data'])` and the PHP notice disappeared.

Comment (fedys) [reply Form/Type/CampaignConditionFieldValueType.php]: That's really strange. I am unable to reproduce it with this simple code snippet.

```php
<?php

declare(strict_types=1);
error_reporting(-1);

$options = [];
$fields = [];

var_dump(isset($options['data']['field']) && isset($fields[$options['data']['field']]));
```

Comment (escopecz) [reply Form/Type/CampaignConditionFieldValueType.php]: We had a zoom with Fedy. It was OPcode cache issue. It can be simplified as Fedy suggests.

**Review by fedys — APPROVED** [review]
Thanks @escopecz!

**Review by rohitp19 — APPROVED** [review]
Thanks 👍

**Pull request merged**
